### PR TITLE
NIFI-8106 Added an "Are you sure ?" prompt when delete a processor group

### PR DESCRIPTION
**Pull request by ahmedshaaban1999**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

NIFI-8106 Added an "Are you sure ?" prompt when delete a processor group

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by mcgilman** [comment]
Will review...

**Review by mcgilman — COMMENTED** [review]
Thanks for the PR @ahmedshaaban1999! A couple of comments on the PR...

- Kind of nit-picky but thought I should throw this out there. Since both requests are the same and only part of the response handling differs, what are your thoughts on looking to reuse as much as possible?
- The proposed solution works great when the user selects a single Process Group and attempts to delete it. However, if the user defines a selection with more than one component that includes Process Group(s) there is no prompt. For instance, if the user selects two Process Groups or selects one Processor and one Process Group there is no confirmation.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-actions.js [hunk 1]
```diff
@@ -1153,42 +1153,75 @@
...
-                        nfCanvasUtils.setURLParameters();
-
-                        // refresh the birdseye
-                        nfBirdseye.refresh();
-                        // inform Angular app values have changed
-                        nfNgBridge.digest();
-                    }).fail(nfErrorHandler.handleAjaxError);
-                } else {
+					
+					if (nfCanvasUtils.isProcessGroup(selection) === true) {
```
Comment (thenatog) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-actions.js:1157]: The whitespacing on this section of code has gone wrong. Can we update the whitespace to align with the rest of the file?

**Comment by ahmedshaaban1999** [comment]
Thanks for the feedback,

@thenatog that's strange, I have been facing some problems with whitespaces recently, I will try to double check them using vi and notepadd++ 

@mcgilman your points are spot on. I tried to tackle both points at first but wasn't successful, though to myself that I can divide the work into two jira tickets. but I agree to get all things done in this ticket. I am currently working on refactoring to make use of the common code as much as possible and at the same time make the message appear when multi-select.

**Comment by ahmedshaaban1999** [comment]
@mcgilman I have a question. In the deletion of a multi-selection, why after the snippet is removed we loop the components of the snippet and remove them one by one ? 

```
 // remove all the non connections in the snippet first
                            components.each(function (ids, type) {
                                if (type !== 'Connection') {
                                    nfCanvasUtils.getComponentByType(type).remove(ids);
                                }
                            });

                            // then remove all the connections
                            if (components.has('Connection')) {
                                nfConnection.remove(components.get('Connection'));
                            }
```

**Pull request closed without merging**